src/java.base/aix/classes/sun/nio/fs/AhafsPoller.java

```diff
@@ -0,0 +1,602 @@
+        // Infrastructure. So the modify event is detected by monitoring
+        // for changes to the individual files in the directory.
+        if (wk.isWatching(StandardWatchEventKinds.ENTRY_MODIFY)) {
+            try {
+                createSubKeys(watchPath, wk);
+            } catch (IOException e) {
+                // See note about returning exceptions below.
+                return new IOException("[AixWatchService] IO error reported during file registration", e);
+            } catch (AixWatchService.FatalException e) {
+                // AbstractPolller checks if return type is insanceof IOE or RuntimeException
+                // and then throws. To ensure client is informed of the error, we
+                // map our internal exception to one checked by the caller and return.
+                return new IOException("[AixWatchService] Error reported during file registration", e);
+            }
+        }
+
+        return wk;
+    }
+
+    /**
+     * Update an existing AixWatchKey registration.
+     * Replace the currently watched events with the new event set,
+     * and perform any necessary housekeeping.
+     */
+    WatchKey reRegister(AixWatchKey wk,
+                        Set<? extends WatchEvent.Kind<?>> newEvents,
+                        WatchEvent.Modifier... modifiers)
+        throws AixWatchService.FatalException
+    {
+        AixWatchKey.TopLevelKey tlk = wk.resolve();
+        if (newEvents.contains(StandardWatchEventKinds.ENTRY_MODIFY) &&
+            !wk.isWatching(StandardWatchEventKinds.ENTRY_MODIFY)) {
+            try {
+                createSubKeys(wk.watchable(), tlk);
+            } catch (IOException e) {
+                throw new AixWatchService.FatalException("[AixWatchService] Error reported during file registration", e);
+            }
+        }
+        if (wk.isWatching(StandardWatchEventKinds.ENTRY_MODIFY) &&
+            !newEvents.contains(StandardWatchEventKinds.ENTRY_MODIFY)) {
+            for (AixWatchKey.SubKey sk : tlk.subKeys()) {
+                tlk.removeSubKey(sk);
+                cleanKeyResources(sk);
+            }
+        }
+
+        tlk.replaceEvents(newEvents);
+        return tlk;
+    }
+
+    /**
+     * Cancel an AixWatchKey.
+     * This method performs two (slightly different) tasks:
+     *
+     * 1. Cancel the given primary key (provided by the client) and
+     *      any subkeys.
+     * 2. Cancel the given subkey (provided as an interal call). Do not
+     *      resolve to TopLevelKey or cancel SubKeys.
+     *
+     */
+    @Override
+    void implCancelKey(WatchKey wk)
+    {
+        // Only cancel AixWatchKeys given out by this WatchService
+        if (!(wk instanceof AixWatchKey))
+            return;
+
+        AixWatchKey awk = (AixWatchKey) wk;
+        awk.invalidate();
+
+        // If cancelling a TopLevelKey, also cancel any SubKeys
+        if(wk instanceof AixWatchKey.TopLevelKey) {
```

Comment:
let's add space after `if`

Reply:
Yes, by convention no blank space before the left `(` is for functions / methods.